lib/util/determine-files.js

```diff
@@ -0,0 +1,81 @@
+let path = require("path");
+let fs = require("fs");
+
+/* Returns a function that determines files to process
+ *
+ * This is useful for pipelines that process a directory of files.
+ * It takes a source directory and optionally a predicate that is used to filter
+ * the determined files. It automatically filters invisible files.
+ *
+ * If the returned function is called without any arguments, it will return a
+ * Promise with an array of all files that matched.
+ * If it is called with an array of paths, it will return a Promise with an
+ * array of all of those files that matched.
+ */
+module.exports = function(source, predicate = (_ => true)) {
+	return files => {
+		return determineFilesToProcess(source, files).
+			then(skipInvisibleFiles).
+			then(fileNames => fileNames.filter(predicate));
+	};
+};
+
+function determineFilesToProcess(source, files) {
+	if(files) {
+		return filesWithinDirectory(source, files);
+	} else {
+		return tree(source);
+	}
+}
+
+function skipInvisibleFiles(files) {
```

Comment:
I believe they're called _hidden_ files?

However, since this convention only applies to  Unix systems, "dotfiles" seems more accurate anyway.